frontend/sozisel-app/src/components/TemplateCreation/EventCreation/EventCreation.tsx

```
@@ -1,57 +1,121 @@
 import "./EventCreation.scss";
 
+import * as yup from "yup";
+
+import { Controller, useForm } from "react-hook-form";
 import {
-  Button,
   FormControl,
   MenuItem,
   Paper,
   Select,
   TextField,
   Typography,
 } from "@material-ui/core";
-import React, { ReactElement, useState } from "react";
+import React, { ReactElement, useEffect, useState } from "react";
 
+import Quiz from "../Modules/Quiz/Quiz";
+import { QuizContextProvider } from "../../../contexts/Quiz/QuizContext";
+import { quizSchema } from "./Schemas";
+import { useEventContext } from "../../../contexts/Event/EventContext";
 import { useTranslation } from "react-i18next";
+import { yupResolver } from "@hookform/resolvers/yup";
+
+const createSchema = (moduleType: string): yup.AnyObjectSchema => {
+  switch (moduleType) {
+    case "Quiz":
+      return quizSchema;
+    default:
```

I would throw the error on default. This should never happen.